addons/stock/models/stock_move_line.py

```diff
@@ -517,8 +517,8 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                         precision_rounding=self.product_uom_id.rounding,
                         rounding_method='UP')
                     candidate.product_uom_qty = self.product_id.uom_id._compute_quantity(quantity_split, candidate.product_uom_id, rounding_method='HALF-UP')
-                    quantity -= quantity_split
                     move_to_recompute_state |= candidate.move_id
+                    break
                 if quantity == 0.0:
                     break
```

Comment:
maybe ove this condition in the previous if, it may be clearer for the casual reader :)